src/energyplus/ForwardTranslator.cpp

```diff
@@ -4198,24 +4201,22 @@ namespace energyplus {
     }
 
     // ensure at least one life cycle cost exists to prevent crash in E+ 8
-    unsigned numCosts = 0;
-    for (const IdfObject& object : m_idfObjects) {
-      if (object.iddObject().type() == openstudio::IddObjectType::LifeCycleCost_NonrecurringCost) {
-        numCosts += 1;
-      } else if (object.iddObject().type() == openstudio::IddObjectType::LifeCycleCost_RecurringCosts) {
-        numCosts += 1;
+    if (!m_excludeLCCObjects) {
+      bool hasAtLeastOneCost = std::any_of(m_idfObjects.cbegin(), m_idfObjects.cend(), [](const auto& obj) {
+        auto iddObjType = obj.iddObject().type();
+        return (iddObjType == openstudio::IddObjectType::LifeCycleCost_NonrecurringCost)
+               || (iddObjType == openstudio::IddObjectType::LifeCycleCost_RecurringCosts);
+      });
+
+      if (!hasAtLeastOneCost) {
+        // add default cost
+        auto& idfObject = m_idfObjects.emplace_back(openstudio::IddObjectType::LifeCycleCost_NonrecurringCost);
+        idfObject.setString(LifeCycleCost_NonrecurringCostFields::Name, "Default Cost");
+        idfObject.setString(LifeCycleCost_NonrecurringCostFields::Category, "Construction");
+        idfObject.setDouble(LifeCycleCost_NonrecurringCostFields::Cost, 0.0);
+        idfObject.setString(LifeCycleCost_NonrecurringCostFields::StartofCosts, "ServicePeriod");
```

Comment:
Fix #4404  + use std::any_of instead of looping on all objects (and not even breaking as soon as one is found).
Also use emplace_back to construct in place for performance reasons